index.js

```diff
@@ -61,17 +62,28 @@ var WinstonCloudWatch = function(options) {
 
 util.inherits(WinstonCloudWatch, winston.Transport);
 
-WinstonCloudWatch.prototype.log = function(level, msg, meta, callback) {
-  debug('log (called by winston)', level, msg, meta);
-
-  var log = { level: level, msg: msg, meta: meta };
-  if (!_.isEmpty(msg)) {
+WinstonCloudWatch.prototype.log = function (level, msg, meta, callback) {
+  // Winston 3.x support
```

Comment:
I don't know if keeping retro compatibility is the best thing, winston-cloudwatch will have a new major version, so there shouldn't be any problem with people seeing their code suddenly not working anymore... but yeah I don't like to force people to update their dependencies too.

It just feels a burden that will have to be carried (and remembered) in the future.

Reply:
@lazywithclass Agree. If you cut a new major version, then you can skip this backwards compatibility thing entirely. Then again, that comes with a prize tag. And keep in mind that Winston 3 is still RC.

It's your call.

Reply:
Yes.
I would say let's pause until Winston 3 comes out as stable, then I will come back to this and push out a new major version without retro compatibility.

Reply:
@lazywithclass Want me to update the PR, getting rid of the backwards compatibility logic?

Reply:
@nfantone that would be great, yes

Reply:
@lazywithclass I'm not too user about this `meta` thing in Winston 3. There's no such argument any more. Instead, we get an "info" object containing everything:

```json
{ message: "Hello world", level: "info", foo: "bar", ... }
```

Am I supposed to understand that everything other than `"message"` and `"level"` falls under `"meta"`, now?

Reply:
@nfantone it looks like [they've put everything under what they call `info`](https://github.com/winstonjs/winston#adding-custom-transports); I think we should go with whatever interface they are expecting for the `log` function, if we buy into the non-retro compatibility idea then we should follow it.

If any user was [using this feature](https://github.com/lazywithclass/winston-cloudwatch/blob/12e286b5633533e8c8d05ff685e7d5a5f2119d77/index.js#L23) and now for some reason there's no way to add `meta` to a log line - I think there will be an easy way though - then we will add it again if and when we get a complaint, as long as we are clear about what the breaking changes are in the release notes I don't think there will be major issues.

This said I would wait and re-evaluate whatever decision we arrive to when they release the stable.

Reply:
@lazywithclass So you're basically proposing that we remove it from the library.

Reply:
@nfantone yes because we also give the option of using `options.messageFormatter` if someone want to craft a particular message

Reply:
@lazywithclass Done! Removed compatibility logic, updated tests and `options.messageFormatter`.